extensions/spark/kyuubi-spark-lineage/src/main/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParseHelper.scala

```diff
@@ -301,6 +318,20 @@ trait LineageParser {
       case p: LocalRelation =>
         joinRelationColumnLineage(parentColumnsLineage, p.output, Seq(LOCAL_TABLE_IDENTIFIER))
 
+      case p: InMemoryRelation if p.cacheBuilder.tableName.nonEmpty =>
+        val tableName = p.cacheBuilder.tableName.get
+        // get logical plan from cachedPlan's children
+        p.cacheBuilder.cachedPlan.children.map { sparkPlan =>
+          sparkPlan.logicalLink match {
+            case Some(logicPlan) =>
+              val relationColumnLineage =
+                extractColumnsLineage(logicPlan, ListMap[Attribute, AttributeSet]())
+              mergeRelationColumnLineage(parentColumnsLineage, p.output, relationColumnLineage)
+            case _ =>
```

Comment:
we should find the logic link recursively

Reply:
Iterate through the nodes of the sparkPlan until we find the node where the logicalLink exists, then start parsing it?